src/GraphQLParser/LexerContext.cs

```diff
@@ -402,10 +447,12 @@ private Token ReadName()
 
             do
             {
-                _currentIndex++;
-                code = GetCode();
+                if (++_currentIndex < _source.Length)
+                    code = _source.Span[_currentIndex];
+                else
+                    break;
             }
-            while (IsNotAtTheEndOfQuery() && IsValidNameCharacter(code));
+            while (code == '_' || char.IsLetterOrDigit(code));
```

Comment:
```suggestion
            while (code == '_' || ('a' <= code && code <= 'z') || ('A' <= code && code <= 'Z') || ('0' <= code & code <= '9'));
```
This is faster and meets the spec exactly.

Reply:
Did you specifically write & instead of &&?

Reply:
No, it was an accident.  But a single ampersand might be faster, due to less conditional jump statements and possibly less chance of branch prediction errors in the CPU pipeline.  I have not tested it so I've been just using &&

Reply:
I thought about it too, but didn't use.